src/plugins/intel_gpu/src/plugin/transformations/convert_reducemax_scalar_output.cpp

```diff
@@ -0,0 +1,92 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "convert_reducemax_scalar_output.hpp"
+
+#include <algorithm>
+#include <memory>
+#include <vector>
+
+#include "openvino/op/reduce_max.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/core/rt_info.hpp"
+
+ov::intel_gpu::ConvertReduceMaxScalarOutput::ConvertReduceMaxScalarOutput() {
+    // Check all Reduce nodes
+    auto m = std::make_shared<ov::pass::pattern::Matcher>(ov::pass::pattern::wrap_type<ov::op::v1::ReduceMax>(),
+                                                          "ConvertReduceMaxScalarOutput");
+    register_matcher(m, [&](ov::pass::pattern::Matcher& m) {
+        auto reduce_max = std::dynamic_pointer_cast<ov::op::v1::ReduceMax>(m.get_match_root());
+        if (!reduce_max || transformation_callback(reduce_max)) {
+            return false;
+        }
+
+        const auto input_shape = reduce_max->input_value(0).get_partial_shape();
+        auto reduce_shape = reduce_max->input_value(1).get_partial_shape();
+        if (reduce_shape.is_dynamic() || reduce_shape.size() != 1 || reduce_shape.to_shape()[0] != input_shape.size() ||
+            reduce_shape.to_shape()[0] <= 1) {
+            return false;
```

Comment:
Please move all applicability checks to predicate for reduce op

Reply:
done